packages/stripe_android/android/src/main/kotlin/com/facebook/react/bridge/ReadableMap.java

```diff
@@ -81,7 +81,7 @@ public ReadableType getType(String key) {
                 return ReadableType.Array;
             } else if (value instanceof Number) {
                 return ReadableType.Number;
-            } else if (value instanceof Map) {
+            } else if (value instanceof Map || (value instanceof  JSONObject)) {
```

Comment:
@jonasbark  this is the most tricky fix. I did some tests and functionality still seems to work